modules/navier_stokes/src/physics/WCNSFVTurbulencePhysics.C

```diff
@@ -0,0 +1,328 @@
+                             "von_karman_const",
+                             "von_karman_const_0",
+                             "mixing_length_two_term_bc_expansion"});
+}
+
+void
+WCNSFVTurbulencePhysics::addNonlinearVariables()
+{
+  if (_turbulence_model == "mixing-length")
+    return;
+}
+
+void
+WCNSFVTurbulencePhysics::addAuxiliaryVariables()
+{
+  if (_turbulence_model == "mixing-length")
+  {
+    auto params = getFactory().getValidParams("MooseVariableFVReal");
+    assignBlocks(params, _blocks);
+    if (isParamValid("mixing_length_two_term_bc_expansion"))
+      params.set<bool>("two_term_boundary_expansion") =
+          getParam<bool>("mixing_length_two_term_bc_expansion");
+    getProblem().addAuxVariable("MooseVariableFVReal", _mixing_length_name, params);
+  }
+}
+
+void
+WCNSFVTurbulencePhysics::addFVKernels()
+{
+  if (_has_flow_equations)
+    addFlowTurbulenceKernels();
+  if (_has_energy_equation)
+    addFluidEnergyTurbulenceKernels();
+  if (_has_scalar_equations)
+    addScalarAdvectionTurbulenceKernels();
+}
+
+void
+WCNSFVTurbulencePhysics::addFlowTurbulenceKernels()
+{
+  if (_turbulence_model == "mixing-length")
+  {
+    const std::string u_names[3] = {"u", "v", "w"};
+    const std::string kernel_type = "INSFVMixingLengthReynoldsStress";
+    InputParameters params = getFactory().getValidParams(kernel_type);
+    assignBlocks(params, _blocks);
+    params.set<MooseFunctorName>(NS::density) = _density_name;
+    params.set<MooseFunctorName>(NS::mixing_length) = _mixing_length_name;
+
+    std::string kernel_name = prefix() + "ins_momentum_mixing_length_reynolds_stress_";
+    if (_porous_medium_treatment)
+      kernel_name = prefix() + "pins_momentum_mixing_length_reynolds_stress_";
+
+    params.set<UserObjectName>("rhie_chow_user_object") = _flow_equations_physics->rhieChowUOName();
+    for (const auto dim_i : make_range(dimension()))
+      params.set<MooseFunctorName>(u_names[dim_i]) = _velocity_names[dim_i];
+
+    for (const auto d : make_range(dimension()))
+    {
+      params.set<NonlinearVariableName>("variable") = _velocity_names[d];
+      params.set<MooseEnum>("momentum_component") = NS::directions[d];
+
+      getProblem().addFVKernel(kernel_type, kernel_name + NS::directions[d], params);
+    }
+  }
+}
+
+void
+WCNSFVTurbulencePhysics::addFluidEnergyTurbulenceKernels()
+{
+  if (_turbulence_model == "mixing-length")
+  {
+    const std::string u_names[3] = {"u", "v", "w"};
+    const std::string kernel_type = "WCNSFVMixingLengthEnergyDiffusion";
+    InputParameters params = getFactory().getValidParams(kernel_type);
+    assignBlocks(params, _blocks);
+    params.set<MooseFunctorName>(NS::density) = _density_name;
+    params.set<MooseFunctorName>(NS::cp) = _fluid_energy_physics->getSpecificHeatName();
+    params.set<MooseFunctorName>(NS::mixing_length) = _mixing_length_name;
+    params.set<Real>("schmidt_number") = getParam<Real>("turbulent_prandtl");
```

Comment:
Why the Schmidt number gets the turbulent Pr? This should be the turbulent Sc number, which is generally quite different than the turbuelnt Pr

Reply:
I'll have to check.

Reply:
this kernel is being used on energy. this is correct